src/components/measureEditor/MeasureEditor.tsx

```diff
@@ -78,18 +80,152 @@ const MeasureEditor = () => {
   const { getUserName } = useOktaTokens();
   const userName = getUserName();
   const canEdit = userName === measure.createdBy;
+  const terminologyServiceApi = useTerminologyServiceApi();
+  const [valuesetMsg, setValuesetMsg] = useState(null);
+
+  const getValueSetErrors = async (
+    valuesetsArray: ElmValueSet[],
+    tgtValue: string
+  ): Promise<ElmTranslationError[]> => {
+    const valuesetsErrorArray: ElmTranslationError[] = [];
+    if (valuesetsArray && tgtValue) {
+      const results = await Promise.allSettled(
+        valuesetsArray.map(async (valueSet, i) => {
+          const oid = getOid(valueSet);
+          await terminologyServiceApi
+            .getValueSet(tgtValue, oid, valueSet.locator)
+            .then((response) => {
+              if (response.errorMsg) {
+                const vsErrorForElmTranslationError: ElmTranslationError =
+                  processValueSetErrorForElmTranslationError(
+                    response.errorMsg.toString(),
+                    valueSet.locator
+                  );
+                valuesetsErrorArray.push(vsErrorForElmTranslationError);
+              }
+            });
+        })
+      );
+      return valuesetsErrorArray;
+    }
+  };
+
+  const getTgt = (): any => {
+    return window.localStorage.getItem("TGT");
+  };
+  const getTgtValue = (tgt: any): any => {
+    let tgtValue = null;
+    if (tgt) {
+      let tgtObjFromLocalStorage = JSON.parse(tgt);
+      for (const [key, value] of Object.entries(tgtObjFromLocalStorage)) {
+        if (key === "TGT") {
+          tgtValue = value.toString();
+        }
+      }
+    }
+    return tgtValue;
+  };
+
+  const getOid = (valueSet: ElmValueSet): string => {
+    const id = valueSet.id.split("").reverse().join("");
+    const index = id.indexOf("/");
+    const oid = id.substr(0, index).split("").reverse().join("");
+
+    return oid;
```

Comment:
I think this can be simplified. there is a regex for oid: https://hl7.org/fhir/datatypes.html#oid
e.g `valueSet.id.match(/[0-2](\.(0|[1-9][0-9]*))+/)[0]` have to check for match(...) results before getting 0th match.